.github/workflows/compatability-tests.yaml

```diff
@@ -0,0 +1,290 @@
+      image_id: ${{ steps.build_export.outputs.ImageID }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16.17.1"
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.2
+        id: go
+
+      - name: Retrieve generated code
+        uses: actions/download-artifact@v3
+        with:
+          name: generated-code
+          path: /tmp/
+
+      - name: Unpack generated code
+        run: tar -xf /tmp/generated.tar.gz
+
+      - name: Extract version
+        shell: bash
+        run: echo "tag=sha-$(git rev-parse --short HEAD | sed s/^v//g)" >> $GITHUB_OUTPUT
+        id: version
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and export
+        uses: docker/build-push-action@v2
+        id: build_export
+        with:
+          context: .
+          file: ./Dockerfile
+          tags: treeverse/lakefs:${{ steps.version.outputs.tag }}
+          outputs: type=docker,dest=/tmp/lakefs.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: lakefs-image
+          path: /tmp/lakefs.tar
+
+  compatibility-checks-client:
+    name: Test lakeFS Hadoop FileSystem compatibility
+    needs: gen-code
+    strategy:
+      fail-fast: false
+      matrix:
+        # Removing a version from this list means the published client is no longer compatible with
+        # that lakeFS version.
+        lakeFS_version: [ 0.89.1, 0.90.0, 0.91.0, 0.92.0 ]
+    runs-on: ubuntu-20.04
+    env:
+      TAG: ${{ matrix.lakeFS_version }}
+      REPO: treeverse
+      SPARK_TAG: 3
+    steps:
+      - name: Check-out code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: "adopt-hotspot"
+          java-version: "8"
+          cache: "sbt"
+
+      - name: Package Spark App
+        working-directory: test/spark/app
+        run: sbt sonnets-311/package
+
+      - name: Generate uniquifying value
+        id: unique
+        run: echo "value=$RANDOM" >> $GITHUB_OUTPUT
+
+      - name: Start lakeFS for Spark tests
+        uses: ./.github/actions/bootstrap-test-lakefs
+        with:
+          compose-directory: test/spark
+        env:
+          REPO: treeverse
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          LAKEFS_DATABASE_TYPE: postgres
+          LAKEFS_BLOCKSTORE_TYPE: s3
+          LAKEFS_BLOCKSTORE_S3_CREDENTIALS_ACCESS_KEY_ID: ${{ secrets.ESTI_AWS_ACCESS_KEY_ID }}
+          LAKEFS_BLOCKSTORE_S3_CREDENTIALS_SECRET_ACCESS_KEY: ${{ secrets.ESTI_AWS_SECRET_ACCESS_KEY }}
+
+      - name: Setup lakeFS for tests
+        working-directory: test/spark
+        run: ./setup-test.sh
+
+      - name: Retrieve client
```

Comment:
spark-submit + --package will download the lakeFSFS for us, why do we need this step?

Reply:
That's a test for the latest client with the server versions. The latest client is not published.